File: templates/matlab/bar_combo.m
function fig = bar_combo()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    apply_theme(); rng(5);
    months = arrayfun(@(i)sprintf('M%d',i),1:12,'UniformOutput',false);
    sales = 30 + 70*rand(1, 12);
    growth = gradient(sales)./sales*100;
    fig = figure;
    yyaxis left;  bar(sales, 'FaceColor', palette('cat',1)); ylabel('sales');
    set(gca,'XTick',1:12,'XTickLabel',months);
    yyaxis right; plot(1:12, growth, '-o', 'Color', palette('cat',2), 'LineWidth', 1.5);
    ylabel('growth (%)'); title('Bar + line combo'); grid on;
end

File: templates/python/bar_combo.py
"""bar_combo: 柱状 + 折线组合（双 Y 轴）."""
import sys, os; sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
import numpy as np
import matplotlib.pyplot as plt
from theme import apply_theme
from palette import cycle

def make_figure(title='Bar + line combo'):
    apply_theme()
    months = [f'M{i+1}' for i in range(12)]
    sales = np.random.default_rng(5).uniform(30, 100, 12)
    growth = np.gradient(sales) / sales * 100
    fig, ax = plt.subplots()
    ax.bar(months, sales, color=cycle(0), alpha=0.85, label='sales')
    ax.set_ylabel('sales', color=cycle(0))
    ax.tick_params(axis='y', labelcolor=cycle(0))
    ax2 = ax.twinx()
    ax2.plot(months, growth, '-o', color=cycle(1), label='growth %')
    ax2.set_ylabel('growth (%)', color=cycle(1))
    ax2.tick_params(axis='y', labelcolor=cycle(1))
    ax2.spines['right'].set_visible(True)
    ax.set_title(title)
    fig.tight_layout()
    return fig

if __name__ == '__main__':
    fig = make_figure(); fig.savefig(__file__.replace('.py', '.png'), dpi=150)
